enterprise/app/filter/filter.tsx

```diff
@@ -123,16 +126,44 @@ export default class FilterComponent extends React.Component<FilterProps, State>
           search.get(MINIMUM_DURATION_PARAM_NAME) ||
           search.get(MAXIMUM_DURATION_PARAM_NAME)
       ),
-      user: search.get(USER_PARAM_NAME),
-      repo: search.get(REPO_PARAM_NAME),
-      branch: search.get(BRANCH_PARAM_NAME),
-      commit: search.get(COMMIT_PARAM_NAME),
-      host: search.get(HOST_PARAM_NAME),
-      command: search.get(COMMAND_PARAM_NAME),
-      minimumDuration: Number(search.get(MINIMUM_DURATION_PARAM_NAME)),
-      maximumDuration: Number(search.get(MAXIMUM_DURATION_PARAM_NAME)),
-      sortBy: search.get(SORT_BY_PARAM_NAME),
-      sortOrder: search.get(SORT_ORDER_PARAM_NAME),
+      user: search.get(USER_PARAM_NAME) || "",
+      repo: search.get(REPO_PARAM_NAME) || "",
+      branch: search.get(BRANCH_PARAM_NAME) || "",
+      commit: search.get(COMMIT_PARAM_NAME) || "",
+      host: search.get(HOST_PARAM_NAME) || "",
+      command: search.get(COMMAND_PARAM_NAME) || "",
+      minimumDuration: Number(search.get(MINIMUM_DURATION_PARAM_NAME)) || undefined,
+      maximumDuration: Number(search.get(MAXIMUM_DURATION_PARAM_NAME)) || undefined,
+      sortBy: search.get(SORT_BY_PARAM_NAME) || "",
+      sortOrder: search.get(SORT_ORDER_PARAM_NAME) || "",
+    };
+  }
+
+  updateFilterState(prevState: State, search: URLSearchParams) {
+    return {
+      isDatePickerOpen: prevState.isDatePickerOpen,
+      isFilterMenuOpen: prevState.isFilterMenuOpen,
+      isSortMenuOpen: prevState.isSortMenuOpen,
```

Comment:
when updating state, the keys which aren't being updated can be omitted:

```suggestion
```

Reply:
Oh weird, I thought these were required to be set if they weren't optional in the State declaration, but it looks like it works fine without setting them, so removed.